packages/frontend/amp/components/elements/Timeline.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Expandable } from '@frontend/amp/components/Expandable';
+import { css } from 'emotion';
+import { palette } from '@guardian/pasteup/palette';
+
+const eventsWrapper = css`
+    margin-left: 8px;
+
+    li:not(:last-child) {
+        border-left: 1px solid rgba(220, 220, 220, 0.5);
+    }
+`;
+
+const eventStyle = css`
+    padding-left: 17px;
+    padding-bottom: 16px;
+`;
+
+const highlight = css`
+    background-color: ${palette.highlight.main};
+`;
+
+const eventIconStyle = css`
+    ${highlight}
+
+    :before {
+        content: '';
+        width: 16px;
+        height: 16px;
+        border-radius: 100%;
+        float: left;
+        margin-left: -25px;
+        background-color: ${palette.neutral[7]};
+    }
+`;
+
+const getHTML = (events: TimelineEvent[], description?: string): string => {
+    // TODO using ternary as && doesn't seem to work here - it prints out undefined
+    const eventMarkup = events.map(e => (
+        <li className={eventStyle}>
+            <time className={eventIconStyle}>{e.date}</time>
```

Comment:
Yeah - at the moment I've just passed them as strings to avoid duplicating the formatting logic from Frontend, but I'd like to pass through an actual date so that we can do this and format as appropriate in DCR.

We can definitely do this now though so shout if you think it's worthwhile. (At the moment I've got a TODO in Frontend around the Timeline case class noting that we should really pass through a richer date object.)

Reply:
I think there's probably a wider questions around machine readable data. I couldn't find a specific `timeline` type reference in Schema but something like https://schema.org/EventSeries might work. I don't know if it's worthwhile now or looking at holistically across everything a bit closer - `datetime` probably doesn't give you a whole lot of advantage alone in this specific case.

Reply:
ah yes definitely. There is probably stuff to look at after we get live here - which I realise means it may never happen - but I'd prefer product to appreciate the cost/prioritise here than mask it in a slower migration.